stage1/init/common/pod.go

```diff
@@ -565,6 +565,50 @@ func PodToSystemd(p *stage1commontypes.Pod, interactive bool, flavor string, pri
 	return nil
 }
 
+// evaluateAppMountPath tries to resolve symlinks within the path.
+// It returns the actual relative path for the given path.
+// TODO(yifan): This is a temporal fix for systemd-nspawn not handling symlink mounts well.
+func evaluateAppMountPath(appRootfs, path string) (string, error) {
+	link := appRootfs
+
+	paths := strings.Split(path, "/")
+	for i, p := range paths {
+		next := filepath.Join(link, p)
+
+		fi, err := os.Lstat(next)
+		if err != nil {
+			if os.IsNotExist(err) {
+				link = filepath.Join(append([]string{link}, paths[i:]...)...)
+				break
+			}
+			return "", err
+		}
+
+		if fi.Mode()&os.ModeType != os.ModeSymlink {
+			link = filepath.Join(link, p)
+			continue
+		}
+
+		// Evaluate the symlink.
+		target, err := os.Readlink(next)
+		if err != nil {
+			return "", err
+		}
+
+		if filepath.IsAbs(target) {
+			link = filepath.Join(appRootfs, target)
```

Comment:
We could use `path.Clean(target)` to ensure that nothing wrong happens if target == "/../foo".
https://golang.org/pkg/path/#Clean